src/main/kotlin/org/jetbrains/research/testspark/tools/ToolUtils.kt

```diff
@@ -11,82 +11,66 @@ import org.jetbrains.research.testspark.core.data.TestGenerationData
-        if (!testSuiteCode.contains("package")) return ""
-        val result = testSuiteCode.replace("\r\n", "\n").split("\n")
-            .filter { it.contains("^package".toRegex()) }.joinToString("").split("package ")[1].split(";")[0]
-        if (result.isBlank()) return ""
-        return result
-    }
-
     /**
      * Saves the data related to test generation in the specified project's workspace.
      *
      * @param project The project in which the test generation data will be saved.
      * @param report The report object to be added to the test generation result list.
-     * @param packageLine The package declaration line of the test generation data.
+     * @param packageName The package declaration line of the test generation data.
      * @param importsCode The import statements code of the test generation data.
      */
     fun saveData(
         project: Project,
         report: Report,
-        packageLine: String,
+        packageName: String,
         importsCode: MutableSet<String>,
         fileUrl: String,
         generatedTestData: TestGenerationData,
+        language: Language = Language.Java,
     ) {
         generatedTestData.fileUrl = fileUrl
-        generatedTestData.packageLine = packageLine
+        generatedTestData.packageName = packageName
         generatedTestData.importsCode.addAll(importsCode)
 
         project.service<TestsExecutionResultService>().initExecutionResult(report.testCaseList.values.map { it.id })
 
         for (testCase in report.testCaseList.values) {
             val code = testCase.testCode
-            testCase.testCode = JavaClassBuilderHelper.generateCode(
-                project,
-                getClassWithTestCaseName(testCase.testName),
-                code,
-                generatedTestData.importsCode,
-                generatedTestData.packageLine,
-                generatedTestData.runWith,
-                generatedTestData.otherInfo,
-                generatedTestData,
-            )
+            testCase.testCode = when (language) {
+                Language.Java -> JavaClassBuilderHelper.generateCode(
+                    project,
+                    getClassWithTestCaseName(testCase.testName),
+                    code,
+                    generatedTestData.importsCode,
+                    generatedTestData.packageName,
+                    generatedTestData.runWith,
+                    generatedTestData.otherInfo,
+                    generatedTestData,
+                )
+
+                Language.Kotlin -> KotlinClassBuilderHelper.generateCode(
+                    project,
+                    getClassWithTestCaseName(testCase.testName),
+                    code,
+                    generatedTestData.importsCode,
+                    generatedTestData.packageName,
+                    generatedTestData.runWith,
+                    generatedTestData.otherInfo,
+                    generatedTestData,
+                )
+            }
```

Comment:
This is the reason why having member methods for a class hierarchy is better than having static methods. If `ClassBuilderHelper.generateCode` methods was not static, the code would look like this:
```kotlin
val myBuilderHelper = when(language) {
  Kotlin -> KotlinClassBuilderHelper()
  Java -> JavaClassBuilderHelper()
}

// ...

for (testCase in report.testCaseList.values) {
  myBuilderHelper.generateCode(
                    project,
                    getClassWithTestCaseName(testCase.testName),
                    testCase.testCode,
                    generatedTestData.importsCode,
                    generatedTestData.packageName,
                    generatedTestData.runWith,
                    generatedTestData.otherInfo,
                    generatedTestData,
  )
}
```
Again, imagine at least one more language supported; calling the same method with the same parameters but on a different class 3 times is not very convenient.

Reply:
https://github.com/JetBrains-Research/TestSpark/issues/301